### PR TITLE
wss_sim: state transition simulator

### PR DESCRIPTION
**Pull request by arnetheduck**
It's sometimes useful to simulate what happens when a chain runs from a
given state with a given set of private keys - `wss_sim` allows running
such a simulation.

One use of such a tool is to simulate a weak subjectivity attack,
creating alternative histories of the same chain:
https://notes.status.im/nimbus-insecura-network#

### CODE REVIEW

File: research/wss_sim.nim [hunk 1]
```diff
@@ -0,0 +1,259 @@
+# beacon_chain
+# Copyright (c) 2019-2021 Status Research & Development GmbH
```
Comment (tersec) [inline research/wss_sim.nim]: 2021?

Comment (arnetheduck) [reply research/wss_sim.nim]: eh right, that's when I had the sim ready :)

**Review by tersec — APPROVED** [review]

**Review by mratsim — APPROVED** [review]
Excellent!

File: research/wss_sim.nim (hunk 1 above)
Comment (arnetheduck) [inline research/wss_sim.nim]:
```suggestion
# Copyright (c) 2022 Status Research & Development GmbH
```

**Pull request merged**